bam_sort.c

```diff
@@ -1103,19 +1105,17 @@ int bam_merge_core2(int by_qname, char* sort_tag, const char *out, const char *m
 
         hdr[i] = hin;
 
+        int order_ok = 1;
         if ((translation_tbl+i)->lost_coord_sort && !by_qname) {
             fprintf(stderr, "[bam_merge_core] Order of targets in file %s caused coordinate sort to be lost\n", fn[i]);
+            order_ok = 0;
         }
 
-        // Potential future improvement is to share headers between CRAM files for
-        // samtools sort (where all headers are identical.
-        // Eg:
-        //
-        // if (i > 1) {
-        //     sam_hdr_free(cram_fd_get_header(fp[i]->fp.cram));
-        //     cram_fd_set_header(fp[i]->fp.cram, cram_fd_get_header(fp[0]->fp.cram));
-        //     sam_hdr_incr_ref(cram_fd_get_header(fp[0]->fp.cram));
-        // }
+        if (!refs && cram_get_refs(fp[i]))
```

Comment:
`if (!refs)` should be enough.

Reply:
Agreed.  Not sure what I was thinking there.